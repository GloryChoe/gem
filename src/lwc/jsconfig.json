{
    "compilerOptions": {
        "experimentalDecorators": true,
        "baseUrl": ".",
        "paths": {
            "c/sge_defaultGau": [
                "sge_defaultGau/sge_defaultGau.js"
            ],
            "c/sge_dynamicForm": [
                "sge_dynamicForm/sge_dynamicForm.js"
            ],
            "c/sge_formField": [
                "sge_formField/sge_formField.js"
            ],
            "c/sge_formSection": [
                "sge_formSection/sge_formSection.js"
            ],
            "c/sge_service": [
                "sge_service/sge_service.js"
            ]
        }
    },
    "include": [
        "**/*",
        "../../.sfdx/typings/lwc/**/*.d.ts"
    ],
    "typeAcquisition": {
        "include": [
            "jest"
        ]
    }
}
