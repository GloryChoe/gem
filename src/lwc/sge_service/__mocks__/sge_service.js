
const SUCCESS = {
    "sObjectName": "Opportunity",
    "sections": [{
        "key": "1798336604",
        "label": "Some Section",
        "columns": [{
            "key": "596148284",
            "fields": [{
                "name": "npsp__Ask_Date__c",
                "label": "Ask Date",
                "helpText": "The date the donor was asked for this Donation. This may be different from the date you received it."
            }, {
                "name": "npsp__In_Kind_Type__c",
                "label": "In-Kind Type",
                "helpText": "The type of In-Kind Gift, for example, Goods or Services."
            }]
        }, {
            "key": "780860961",
            "fields": [{
                "name": "npsp__Fair_Market_Value__c",
                "label": "Fair Market Value",
                "helpText": "The Fair Market Value of an In-Kind gift."
            }]
        }]
    }]
};

// resolve to success by default
export const getOpportunityLayout = jest.fn().mockResolvedValue(SUCCESS);
