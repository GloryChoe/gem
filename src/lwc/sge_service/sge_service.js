import readOpportunityLayout from '@salesforce/apex/SGE_GiftEntryService.readOpportunityLayout';
import readAllocationSettings from '@salesforce/apex/SGE_GiftEntryService.readAllocationSettings';

const getOpportunityLayout = () => {
    return new Promise((resolve, reject) => {
        readOpportunityLayout()
            .then((result) => {
                resolve(result);
            })
            .catch(error => {
                console.error(JSON.stringify(error));
            });
    });
};

const getAllocationSettings = () => {
    return new Promise((resolve, reject) => {
       readAllocationSettings()
           .then(resolve)
           .catch(error => {
               console.error(JSON.stringify(error));
           });
    });
};

export { getOpportunityLayout, getAllocationSettings }