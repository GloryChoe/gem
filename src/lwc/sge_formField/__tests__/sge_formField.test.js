import { createElement } from 'lwc';
import SGE_FormField from 'c/sge_formField';
import { getOpportunityLayout } from 'c/sge_service';

jest.mock('c/sge_service');

describe('c-sge_form-field', () => {

    afterEach(clearDOM);

    it('should load', () => {
        const element = createElement('c-sge_form-field', { is: SGE_FormField });

        document.body.appendChild(element);
    });
});