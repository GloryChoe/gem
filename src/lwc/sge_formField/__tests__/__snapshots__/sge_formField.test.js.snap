// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-sge_form-field should return a field object with a name and value 1`] = `
Object {
  "Test_Number_Field__c": 10,
}
`;
